x-pack/plugins/infra/server/routes/infra/lib/mapper.ts

```diff
@@ -32,16 +34,29 @@ export const mapToApiResponse = (
     };
   }
 
-  const hosts = buckets.map((bucket) => {
-    const metrics = convertMetricBucket(params, bucket);
-    const metadata = convertMetadataBucket(bucket);
+  const hosts = buckets
+    .map((bucket) => {
+      const metrics = convertMetricBucket(params, bucket);
+      const metadata = convertMetadataBucket(bucket);
 
-    return { name: bucket.key as string, metrics, metadata };
-  });
+      return { name: bucket.key as string, metrics, metadata };
+    })
+    .sort((a, b) => {
+      const cpuAValue = a.metrics.find((metric) => metric.name === 'cpu')?.value ?? null;
```

Comment:
I see that it can simplify the sorting but why would we want to add another field `cpuValue` to the node, the cpu is already set in the metrics, it's adding redundant information.

Reply:
It'd be temporary, just to help in the sorting.

We'd remove `cpuValue` from `nodes` here:

```ts
nodes: hosts.map(({ cpuValue, ...rest }) => rest),
```

Reply:
> I would also sort alertsCount on the server, for consistency. Wdyt?

alertsCount is the default sorting on the client side, so the sorting it's going to be made twice, one on the server and when rendering the table. Does it make sense to do it twice?

Reply:
If we could leave the default sorting for the server and the client to be responsible for sorting when users interact with the table, then IMHO it makes sense. 

In the ideal world - if elasticsearch supported sorting by sub aggs -, sorting would happen on the server side and each interaction to sort the table data would trigger a new request.

I think this endpoint approach to sorting is fundamentally wrong due to the limitation above, so I would say it's ok to leave it the way you did it. My suggestion was more for consistency because we're sorting CPU on the server, but alertsCount on the client.

Reply:
Thanks, then I will leave as it is